src/services/RequirementsTxtGenerator.ts

```diff
@@ -0,0 +1,134 @@
+import * as vscode from 'vscode';
+import * as fs from 'node:fs';
+import type { dh as DhcType } from '@deephaven/jsapi-types';
+
+const LOG_TAG = 'VSCODE-REQUIREMENTS-TXT' as const;
+const EXTRACT_REQUIREMENT_REGEX = new RegExp(`^\\[${LOG_TAG}](.*)`);
+const FINALIZE_LOGS_TAG = `[${LOG_TAG}-END]\n`;
+
+/**
+ * Query installed Python package names + versions. Each package is prefixed with
+ * a tag to identify it in the log messages. After all packages have been printed,
+ * a finalizing log message is sent to indicate the end of the list.
+ */
+const REQUIREMENTS_QUERY_TXT = `from importlib.metadata import packages_distributions, version
```

Comment:
Instead of polluting the user's history, create a table with the data and then you can get the table from the command. Roughly:
```python
from deephaven import new_table
from deephaven.column import string_col
from importlib.metadata import packages_distributions, version

installed = {pkg for pkgs in packages_distributions().values() for pkg in pkgs}

__vscode_requirements = new_table([
    string_col("Name", list(installed)),
    string_col("Version", [version(pkg) for pkg in installed])
])
```

Then get the result of the command after you run it, read the table/take a snapshot, generate the requirements.txt, and delete the table after:
```python
del __vscode_requirements
```

Reply:
@mofojed what's the significance of double leading underscore vs single on a table?